src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPInterfaceStatistics.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+
+namespace System.Net.NetworkInformation
+{
+    /// <summary>
+    /// IPInterfaceStatistics provider for Linux. 
+    /// Reads information out of /proc/net/dev and other locations.
+    /// </summary>
+    internal class LinuxIPInterfaceStatistics : IPInterfaceStatistics
+    {
+        // /proc/net/dev
+        // Receive section
+        private uint _bytesReceived;
+        private uint _packetsReceived;
+        private uint _errorsReceived;
+        private uint _incomingPacketsDropped;
+        private uint _fifoBufferErrorsReceived;
+        private uint _packetFramingErrorsReceived;
+        private uint _compressedPacketsReceived;
+        private uint _multicastFramesReceived;
+
+        // Transmit section
+        private uint _bytesTransmitted;
+        private uint _packetsTransmitted;
+        private uint _errorsTransmitted;
+        private uint _outgoingPacketsDropped;
+        private uint _fifoBufferErrorsTransmitted;
+        private uint _collisionsDetected;
+        private uint _carrierLosses;
+        private uint _compressedPacketsTransmitted;
+
+        // From /sys/class/net/<interface>/tx_queue_len
+        private int _transmitQueueLength;
+
+        public LinuxIPInterfaceStatistics(string name)
+        {
+            ParseTable(name);
+            string transmitQueueLengthFilePath = Path.Combine(NetworkFiles.SysClassNetFolder, name, NetworkFiles.TransmitQueueLengthFileName);
+            _transmitQueueLength = int.Parse(File.ReadAllText(transmitQueueLengthFilePath));
+        }
+
+        private void ParseTable(string name)
+        {
+            /* NOTE: This same information can be obtained using rtnetlink:
```

Comment:
I'm not sure of the intent of this comment: is it a TODO to use `rtnetlink` instead?